### ISSUE TITLE
domain login fails with support-diagnostics

### ISSUE THREAD

**Issue by wolframhaussig** [elastic/support-diagnostics#489]
I wanted to look into a problem on our elastic stack using the support diagnostics tool. As we primarily use the active directory login I tried to use the tool with my personal account which has superuser privileges on the elastic cluster. Unfortunately, I get the following error:

```
ERROR com.elastic.support.diagnostics.chain.DiagnosticChainExec - Unexpected error
java.lang.IllegalArgumentException: Illegal character in path at index 21: /_security/user/mydomain\myuser
	at java.net.URI.create(URI.java:883) ~[?:?]
	at org.apache.http.client.methods.HttpGet.<init>(HttpGet.java:66) ~[httpclient-4.5.10.jar:4.5.10]
	at com.elastic.support.rest.RestClient.execGet(RestClient.java:66) ~[support-diagnostics-8.1.4.jar:8.1.4]
	at com.elastic.support.rest.RestClient.execQuery(RestClient.java:58) ~[support-diagnostics-8.1.4.jar:8.1.4]
	at com.elastic.support.diagnostics.commands.CheckUserAuthLevel.execute(CheckUserAuthLevel.java:42) ~[support-diagnostics-8.1.4.jar:8.1.4]
	at com.elastic.support.diagnostics.chain.DiagnosticChainExec.runDiagnostic(DiagnosticChainExec.java:31) [support-diagnostics-8.1.4.jar:8.1.4]
	at com.elastic.support.diagnostics.DiagnosticService.exec(DiagnosticService.java:68) [support-diagnostics-8.1.4.jar:8.1.4]
	at com.elastic.support.diagnostics.DiagnosticApp.main(DiagnosticApp.java:42) [support-diagnostics-8.1.4.jar:8.1.4]
Caused by: java.net.URISyntaxException: Illegal character in path at index 21: /_security/user/mydomain\myuser
	at java.net.URI$Parser.fail(URI.java:2913) ~[?:?]
	at java.net.URI$Parser.checkChars(URI.java:3084) ~[?:?]
	at java.net.URI$Parser.parseHierarchical(URI.java:3166) ~[?:?]
	at java.net.URI$Parser.parse(URI.java:3125) ~[?:?]
	at java.net.URI.<init>(URI.java:600) ~[?:?]
	at java.net.URI.create(URI.java:881) ~[?:?]
	... 7 more
```

Of course I can use the default elastic superuser but my question is: Am I doing something wrong or is the support tool not designed for active directory?

This issue was created after asking the question on the elastic forum here: https://discuss.elastic.co/t/domain-login-with-elastic-diagnostics-tool/271460

**Comment by pickypg** [comment]
We most likely need to URL encode the username for that specific API. The `\` is probably `%5C` (untested to confirm that that fixes it).

I doubt you can do anything locally to work around that because other APIs will expect the unencoded value in the auth header.

**Comment by pickypg** [comment]
I have a PR up to fix this. In the meantime, before it's merged and released, the only workaround is to use a user without a special character in it.
